hubblestack/extmods/hubble_mods/misc.py

```diff
@@ -0,0 +1,1096 @@
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': "/some/path/file.txt", 'status': True},
+                  'caller': 'Audit'}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing misc module for id: {0}'.format(block_id))
+    
+    function_name = runner_utils.get_param_for_module(block_id, block_dict, 'function')
+
+    result = FUNCTION_MAP[function_name](block_id, block_dict, extra_args)
+
+    if result is True:
+        return runner_utils.prepare_positive_result_for_module(block_id, None)
+    return runner_utils.prepare_negative_result_for_module(block_id, result)
+
+def _check_all_ports_firewall_rules(block_id, block_dict, extra_args):
+    """
+    Ensure firewall rule for all open ports
+    """
+    start_open_ports = (_execute_shell_command('netstat -ln | grep "Active Internet connections (only servers)" -n | cut -d ":" -f1', python_shell=True)).strip()
+    end_open_ports = (_execute_shell_command('netstat -ln | grep "Active UNIX domain sockets (only servers)" -n  | cut -d ":" -f1', python_shell=True)).strip()
+    open_ports = (_execute_shell_command('netstat -ln | awk \'FNR > ' + start_open_ports + ' && FNR < ' + end_open_ports + ' && $6 == "LISTEN" && $4 !~ /127.0.0.1/ {print $4}\' | sed -e "s/.*://"', python_shell=True)).strip()
+    open_ports = open_ports.split('\n') if open_ports != "" else []
+    firewall_ports = (_execute_shell_command('iptables -L INPUT -v -n | awk \'FNR > 2 && $11 != "" && $11 ~ /^dpt:/ {print $11}\' | sed -e "s/.*://"', python_shell=True)).strip()
+    firewall_ports = firewall_ports.split('\n') if firewall_ports != "" else []
+    no_firewall_ports = []
+
+    for open_port in open_ports:
+        if open_port not in firewall_ports:
+            no_firewall_ports.append(open_port)
+
+    return True if len(no_firewall_ports) == 0 else str(no_firewall_ports)
+
+def _check_password_fields_not_empty(block_id, block_dict, extra_args):
+    """
+    Ensure password fields are not empty
+    """
+    result = _execute_shell_command('cat /etc/shadow | awk -F: \'($2 == "" ) { print $1 " does not have a password "}\'', python_shell=True)
+    return True if result == '' else result
+
+def _ungrouped_files_or_dir(block_id, block_dict, extra_args):
+    """
+    Ensure no ungrouped files or directories exist
+    """
+    raise CommandExecutionError('Module disabled due to performance concerns')
+    result = _execute_shell_command('df --local -P | awk {\'if (NR!=1) print $6\'} | xargs -I \'{}\' find \'{}\' -xdev -nogroup', python_shell=True)
+    return True if result == '' else result
+
+
+def _unowned_files_or_dir(block_id, block_dict, extra_args):
+    """
+    Ensure no unowned files or directories exist
+    """
+    raise CommandExecutionError('Module disabled due to performance concerns')
```

Comment:
No need of any implementation if this module is disabled. Infact we can remove this functionality altogether from this module.